ale_linters/go/gometalinter.vim

```diff
@@ -14,7 +14,6 @@ function! ale_linters#go#gometalinter#GetCommand(buffer) abort
     let l:options = ale#Var(a:buffer, 'go_gometalinter_options')
 
     return ale#Escape(l:executable)
-    \   . ' --include=' . ale#Escape('^' . ale#util#EscapePCRE(l:filename))
```

Comment:
I believe this was added no only for filtering the errors, but because it executes much more quickly.

Could you make showing errors for other files optional?

Reply:
I can make it optional, if it needs to be, but I ran a benchmark on the two executions and they came out just about the same on average...
averages are as follows
```
plain gometalinter call: 4346553250.25 ns/op
gometalinter with include: 4325110610.25 ns/op
```
```
BenchmarkPlain-4               1        4972665842 ns/op
BenchmarkPlain-4               1        4349992730 ns/op
BenchmarkPlain-4               1        4421776655 ns/op
BenchmarkPlain-4               1        4475188394 ns/op
BenchmarkPlain-4               1        4408009068 ns/op
BenchmarkPlain-4               1        4233487406 ns/op
BenchmarkPlain-4               1        4878184255 ns/op
BenchmarkPlain-4               1        4374577545 ns/op
BenchmarkPlain-4               1        4025479863 ns/op
BenchmarkPlain-4               1        4355798558 ns/op
BenchmarkPlain-4               1        4199595247 ns/op
BenchmarkPlain-4               1        4376063551 ns/op
BenchmarkPlain-4               1        4216004173 ns/op
BenchmarkPlain-4               1        3993578021 ns/op
BenchmarkPlain-4               1        4078005741 ns/op
BenchmarkPlain-4               1        4672426812 ns/op
BenchmarkPlain-4               1        4293836100 ns/op
BenchmarkPlain-4               1        4188911073 ns/op
BenchmarkPlain-4               1        4047435325 ns/op
BenchmarkPlain-4               1        4370048646 ns/op
BenchmarkInclude-4             1        4610757981 ns/op
BenchmarkInclude-4             1        4016014938 ns/op
BenchmarkInclude-4             1        4065846142 ns/op
BenchmarkInclude-4             1        4131692572 ns/op
BenchmarkInclude-4             1        4991080619 ns/op
BenchmarkInclude-4             1        4437625405 ns/op
BenchmarkInclude-4             1        4512699377 ns/op
BenchmarkInclude-4             1        4823607824 ns/op
BenchmarkInclude-4             1        4585688367 ns/op
BenchmarkInclude-4             1        4178835349 ns/op
BenchmarkInclude-4             1        3855721441 ns/op
BenchmarkInclude-4             1        3873105094 ns/op
BenchmarkInclude-4             1        4303892553 ns/op
BenchmarkInclude-4             1        4838156426 ns/op
BenchmarkInclude-4             1        3950300377 ns/op
BenchmarkInclude-4             1        3852298474 ns/op
BenchmarkInclude-4             1        4937783666 ns/op
BenchmarkInclude-4             1        4167373118 ns/op
BenchmarkInclude-4             1        4063617353 ns/op
BenchmarkInclude-4             1        4306115129 ns/op
```

```
func BenchmarkPlain(b *testing.B) {
	for n := 0; n < b.N; n++ {
		cmd := exec.Command("sh", "-c", "gometalinter")
		_, _ = cmd.CombinedOutput()
	}
}

func BenchmarkInclude(b *testing.B) {
	for n := 0; n < b.N; n++ {
		cmd := exec.Command("sh", "-c", "gometalinter", "--include main.go")
		_, _ = cmd.CombinedOutput()
	}
}
```

Reply:
@w0rp as far as I can tell, all that include does is run a regex on the results, so it shouldn't be affecting execution time much and maybe even slowing it down if there are tons of errors that need to be regexed

https://github.com/alecthomas/gometalinter/blob/9165748669edf1f7e77f96f1e01b01aa24a26ad3/execute.go#L269

Reply:
Okay sounds good. You can just remove it then. If someone complains about it, we can handle it later.